src/backend/src/services/projects.services.ts

```diff
@@ -1071,4 +1071,57 @@ export default class ProjectsService {
 
     return { ...newUnit, materials: [] };
   }
+
+  /**
+   * Updates the linkType's name, iconName, or required.
+   * @param linkTypeId the current name/id of the linkType
+   * @param name the new name/id of the linkType
+   * @param iconName the new iconName
+   * @param required the new required status
+   * @param submitter user requesting the edit
+   */
+  static async editLinkType(linkTypeId: string, name: string, iconName: string, required: boolean, submitter: User) {
+    if (!isHead(submitter.role)) throw new AccessDeniedException('Only the head or admin can update the linkType');
+
+    // check if the linkType we are trying to update exists
+    const linkType = await prisma.linkType.findUnique({
+      where: { name: linkTypeId }
+    });
+
+    if (!linkType) throw new NotFoundException('Link Type', linkTypeId);
+
+    // check if the new name already exists in the databse
+    const existingLinkType = await prisma.linkType.findUnique({
+      where: { name }
+    });
+    if (existingLinkType && existingLinkType.name !== linkTypeId) {
+      throw new Error(`A linkType with the name '${name}' already exists.`);
+    }
+
+    let linkTypeUpdated: Prisma.LinkTypeGetPayload<typeof linkTypeQueryArgs> | null = null;
```

Comment:
ok so, I realized looking through here, we don't want name in the payload since we're using it as the id. Not having to update the name should simplify a lot of this code (see how we do editing for smthn like Vendor using prisma)